packaging_qa/Debian.bullseye.dockerfile

```diff
@@ -0,0 +1,62 @@
+# Use Debian as the base image
+FROM debian:bullseye
+
+# Set environment variables to avoid interactive prompts
+ENV DEBIAN_FRONTEND=noninteractive
+ARG VERSION=13.0.1
+ENV CITUS_VERSION=${VERSION}.citus-1
+ENV PG_MAJOR=15
+
+# Install dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release \
+    sudo
+
+# Add PostgreSQL repository
+RUN curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | tee /etc/apt/trusted.gpg.d/postgresql.asc && \
+    echo "deb http://apt.postgresql.org/pub/repos/apt bullseye-pgdg main" | tee /etc/apt/sources.list.d/pgdg.list
+
+# Install PostgreSQL $PG_MAJOR and required packages
+RUN apt-get update && apt-get install -y postgresql-$PG_MAJOR postgresql-contrib
+
+# Install Citus repository and required extensions
+RUN curl -s https://install.citusdata.com/community/deb.sh | bash && \
+    apt-get update && apt-get install -y \
+        postgresql-$PG_MAJOR-citus-13.0=$CITUS_VERSION \
+        postgresql-$PG_MAJOR-hll=2.18.citus-1 \
+        postgresql-$PG_MAJOR-topn=2.7.0.citus-1
+
+# Ensure PostgreSQL data directory exists and has correct permissions
+RUN mkdir -p /var/lib/postgresql/$PG_MAJOR/main && chown -R postgres:postgres /var/lib/postgresql && chmod -R 700 /var/lib/postgresql
+
+# Switch to postgres user before running PostgreSQL
+USER postgres
+
+# Initialize the database only if it's not already initialized
```

Comment:
why can we have the database already initialized?

Reply:
To make this more robust, we can check for `pg_control `in `/var/lib/postgresql/${PG_MAJOR}/main/global/`, which ensures that the database directory is actually valid before skipping initialization.

```
RUN if [ ! -f "/var/lib/postgresql/${PG_MAJOR}/main/global/pg_control" ]; then \
       echo "Initializing new PostgreSQL database..."; \
       /usr/lib/postgresql/${PG_MAJOR}/bin/initdb -D /var/lib/postgresql/${PG_MAJOR}/main; \
    else \
       echo "Existing PostgreSQL database detected, skipping initdb."; \
    fi
```

Let me know if you think this approach works, and I can push an update!